pkg/api/validation/validation.go

```diff
@@ -1792,6 +1791,33 @@ func validateProbe(probe *api.Probe, fldPath *field.Path) field.ErrorList {
 	return allErrs
 }
 
+func validateClientIPAffinityConfig(config *api.SessionAffinityConfig, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if config == nil {
+		allErrs = append(allErrs, field.Required(fldPath, fmt.Sprintf("is required when session affinity type is %s", api.ServiceAffinityClientIP)))
```

Comment:
I keep the message `"when when session affinity type is %s", api.ServiceAffinityClientIP` because the field is only required when session affinity type is ClientIP.